packages/styleguide/tsconfig.json

```diff
@@ -17,10 +17,8 @@
     "moduleResolution": "node",
     "resolveJsonModule": true,
     "isolatedModules": true,
-    "jsx": "preserve",
-    "allowSyntheticDefaultImports": true,
-    "declaration": true,
-    "declarationDir": "dist/types"
+    "jsx": "react",
```

Comment:
I think the correct value to use with React 17 would be `"react-jsx"`.
https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/#react-17-jsx-factories